pkg/server/jwk_resolver.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+
+	"github.com/abcxyz/pkg/cache"
+)
+
+type JWKResolver interface {
+	ResolveKeySet(ctx context.Context, oidcHeader string) (jwk.Set, error)
+}
+
+type OIDCResolver struct {
+	cache           *jwk.Cache
+	issuerToJwksURI *cache.Cache[string]
+}
+
+type OpenIDConfiguration struct {
+	// The only field we need from the config
+	JwksURI string `json:"jwks_uri"`
+}
+
+func NewOIDCResolver(ctx context.Context) *OIDCResolver {
+	return &OIDCResolver{
+		cache: jwk.NewCache(ctx),
+		// 24 hour cache timeout, jwks_uri should change very infrequently
+		issuerToJwksURI: cache.New[string](time.Hour * 24),
```

Comment:
I recommend settings this to something smaller like 4 hours for now.

Reply:
Can we make this configurable too from the command line with a default?

Reply:
Added a config option from command line from this that defaults to 4 hours